### PR TITLE
Network config Enhancement

### PR DESCRIPTION
**Pull request by smruti77**
create separate function for all the comparing test and removed
link test,because its already there in ethtool test.

Signed-off-by: bismurti bidhibrata pattajoshi <bbidhibr@in.ibm.com>

### CODE REVIEW

File: io/net/Network_config.py [hunk 1]
```diff
@@ -43,24 +44,27 @@ def setUp(self):
...
-        businfo = process.system_output(cmd, shell=True).strip()
-        self.log.info(businfo)
         cmd = "lspci -v"
         bus_info = process.system_output(cmd, shell=True).strip()
         bus_info = bus_info.split('\n\n')
-        self.log.info("Performing driver match check using lspci and ethtool")
-        self.log.info("-----------------------------------------------------")
         for value in bus_info:
-            if value.startswith(businfo):
+            if value.startswith(self.businfo):
```
Comment (narasimhan-v) [inline io/net/Network_config.py]: Use self.driver here, instead of finding again.

File: io/net/Network_config.py [hunk 2]
```diff
@@ -69,48 +73,60 @@ def test_networkconfig(self):
...
+    def test_speed_check(self):
+        '''
+        Comparing speed
+        '''
         cmd = r"cat /sys/module/%s/drivers/pci\:%s/%s/net/%s/speed" %\
-              (driver, driver, businfo, self.iface)
+              (self.driver, self.driver, self.businfo, self.iface)
         speed = process.system_output(cmd, shell=True).strip()
-        self.log.info("speed is %s" % speed)
-        cmd = "ethtool %s | grep Speed | awk '{print $2}'" % self.iface
```
Comment (narasimhan-v) [inline io/net/Network_config.py]: this command needs to be used, and needs to be used similar to Duplex.

File: io/net/Network_config.py [hunk 3]
```diff
@@ -69,48 +73,60 @@ def test_networkconfig(self):
...
-        self.log.info("Performing Ethtool and interface checks for interface")
-        self.log.info("-----------------------------------------------------")
         if speed != eth_speed:
             self.fail("mis match in speed")
+
+    def test_mac_aadr_check(self):
+        '''
+        comparing mac address
+        '''
+        cmd = r"cat /sys/module/%s/drivers/pci\:%s/%s/net/%s/address" %\
```
Comment (narasimhan-v) [inline io/net/Network_config.py]: These `cmd` is similar for all tests, can we have a function to do this ?

**Comment by smruti77** [comment]
JOB ID     : d98f6ef753b43b0df7fce303730f9884a92daac1
JOB LOG    : /root/sim/avocado-fvt-wrapper/results/job-2019-07-15T15.19-d98f6ef/job.log
 (1/5) Network_config.py:NetworkconfigTest.test_driver_check;Parameters-a107: PASS (0.60 s)
 (2/5) Network_config.py:NetworkconfigTest.test_mtu_check;Parameters-a107: PASS (0.63 s)
 (3/5) Network_config.py:NetworkconfigTest.test_speed_check;Parameters-a107: PASS (0.65 s)
 (4/5) Network_config.py:NetworkconfigTest.test_mac_aadr_check;Parameters-a107: PASS (0.54 s)
 (5/5) Network_config.py:NetworkconfigTest.test_duplex_check;Parameters-a107: PASS (0.63 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Pull request merged**